### ISSUE TITLE
TypeError: path must be a string

### ISSUE THREAD

**Issue by nicksergeant** [AtomLinter/linter-haml#58]
Not sure what's causing this issue, but when I linter-haml runs on a .eco.haml file, I get the following errors:

![http://i.nick.sg/0w3f0u0g222P/Image%202016-04-01%20at%2011.17.42%20AM.png](http://i.nick.sg/0w3f0u0g222P/Image%202016-04-01%20at%2011.17.42%20AM.png)

I'd be happy to supply more information if it might help to debug, as I'm fairly new to Atom. Let me know what that information might be and I'll send along.

**Comment by nicksergeant** [comment]
FWIW, running haml-lint on the file in terminal works fine, but I am seeing this error when I run haml-lint:

```> haml-lint
warning: parser/current is loading parser/ruby21, which recognizes
warning: 2.1.8-compliant syntax, but you are running 2.1.5.
warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
```

**Comment by Arcanemagus** [comment]
Sorry for the delay in replying.

It looks like the author of this failed to specify a default for the "globalHamlLintYmlFile" setting, and since it seems that it couldn't find a configuration file it tried to use the setting... and triggered this error.

**Comment by Arcanemagus** [comment]
This was fixed in v1.2.1.

**Comment by reubenbrown** [comment]
@Arcanemagus just updated to version 2.0.0 and got this same error as well. Seems related to #87

**Comment by Arcanemagus** [comment]
@reubenbrown If you got this error it is definitely _not_ this issue. It looks like @nicksergeant hosted the image on his own site instead of letting GitHub do the hosting so it has been lost, but this was due to the configuration issue that was fixed back in v1.2.1.

If you see this again please file a new issue with the stack trace from the error message.

**Comment by reubenbrown** [comment]
@Arcanemagus Apologies should have clarified more in my original comment, will explain further in a new issue
